test/io/ferguson/readThis/readclass5.chpl

```diff
@@ -7,21 +7,21 @@ class mything {
     this.x = x;
     this.y = y;
   }
-  proc init(r: fileReader) {
+  proc init(r: fileReader(?)) {
```

Comment:
I think that's an interesting question, but specifically for `fileReader`, I don't think the documentation can or should hide the fact that it's generic. You need to be able to talk about the type constructor, e.g. `var x: fileReader(locking=true)`.

I think we could consider making `fileReader` generic-with-defaults, though; so that it defaults to `locking=true` or `locking=false`. It looks like the other fields already have a default.

Reply:
I'd be okay with making it "generic-with-defaults".  And I'd also be okay with the introduction of private fields meaning that the compiler won't yell about specifying privately-generic types without the `(?)` in the future, though I suspect that would complicate the implementation

Reply:
Actually, on second thought, it seems that if we make fileReader generic-with-defaults, then it will be a common issue that you can't read/write a type with a non-locking fileReader (or whatever the non-default type is). The same could be said for other ways it might vary (e.g. what Deserializer is in use).

Another approach is to move the `locking` field from `param` to `const`. I don't remember if we considered that. But, we would run into the above concern with varying Deserializers.